File: frontend/src/modules/verification/permissions.ts
export const VERIFICATION_PERMISSIONS = {
  PROGRESS_CLAIM_READ: "progress_claim:read",
  PROGRESS_CLAIM_CREATE: "progress_claim:create",
  PROGRESS_CLAIM_UPDATE: "progress_claim:update",
  PROGRESS_CLAIM_SUBMIT: "progress_claim:submit",
  PROGRESS_CLAIM_REVIEW: "progress_claim:review",
  PHOTO_BOQ_LINK_READ: "photo_boq_link:read",
  PHOTO_BOQ_LINK_MANAGE: "photo_boq_link:manage",
} as const;

export type VerificationPermission =
  (typeof VERIFICATION_PERMISSIONS)[keyof typeof VERIFICATION_PERMISSIONS];

export function hasVerificationPermission(
  permissions: string[] | undefined,
  permission: VerificationPermission,
): boolean {
  return permissions?.includes(permission) ?? false;
}

File: backend/app/modules/verification/permissions.py
from app.modules.identity.enums import PermissionKey

VERIFICATION_PERMISSIONS = {
  PermissionKey.PROGRESS_CLAIM_READ: (
    "View project progress claims and verification evidence."
  ),
  PermissionKey.PROGRESS_CLAIM_CREATE: (
    "Create project progress claims."
  ),
  PermissionKey.PROGRESS_CLAIM_UPDATE: (
    "Update draft progress claims."
  ),
  PermissionKey.PROGRESS_CLAIM_SUBMIT: (
    "Submit progress claims for verification."
  ),
  PermissionKey.PROGRESS_CLAIM_REVIEW: (
    "Review and approve or reject progress claims."
  ),
  PermissionKey.PHOTO_BOQ_LINK_READ: (
    "View photo to BOQ verification links."
  ),
  PermissionKey.PHOTO_BOQ_LINK_MANAGE: (
    "Create and remove photo to BOQ verification links."
  ),
}